testutil/helpers.go

```diff
@@ -157,3 +158,9 @@ func SetContext(t TestingT, ctx context.Context) {
 func CleanupContext(t *testing.T) {
 	testContexts.Delete(t)
 }
+
+// such as a distro package build.
+func SkipWhenUnprivileged(t *testing.T) {
+	skip.If(t, os.Getuid() != 0, "requires root")
```

Comment:
I'm not actually a Go programmer, I'm just a package maintainer. Someone who presumably knows Go picked this as the method for skipping tests in the patch I inherited in the packaging patchset. If there's a different skip method you'd prefer, I'm all for it.

Reply:
I'm fairly sure this patch set was originally written by @jamieand and `t.Helper()` didn't exist then.  I think it'd be appropriate to add here.

Reply:
> I think it'd be appropriate to add here.

Actually, yes, I think we should; didn't think of `t.Skip` itself printing the line number.

- Without `t.Helper()` it would always print the line-number of this utility; https://go.dev/play/p/20fEoZK6dQL
- With `t.Helper()` it correctly prints the line number of the _call site_; https://go.dev/play/p/PWFYFrgvTDv


One thing I'm considering now; perhaps we should have a `reason string`, so that we can add context to the `t.Skip` (if we know?) For those where we "don't know", we can still pass a blanket `requires root` as message;

```go
testutil.SkipWhenUnprivileged(t, "requires root")
```


As some of the discussions on this PR already show https://github.com/moby/moby/pull/46430/files#r1319245399, sometimes it's not clear what the reason is, and for some, it may actually be "unintentional" for them to not work without root (in which case _ideally_ the message would be a "TODO" or "FIXME" outlining that it "should" work, but the test needs fixing.

Reply:
I added the ability to specify a reason string that defaults to requires root (and rebased), I think this is what you described? Let me know if it needs work. Thanks!